addons/Trusts/Trusts.lua

```diff
@@ -0,0 +1,338 @@
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL from20020516 BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+_addon.name='Trusts'
+_addon.author='from20020516'
+_addon.version='1.1'
+_addon.commands={'trusts','tru'}
+
+config = require('config')
+math = require('math')
+math.randomseed(os.clock())
+require('logger')
+
+windower.register_event('load',function()
+  defaults = {
+    auto=true,
+    language=windower.ffxi.get_info().language,
+    sets={
+      ['default']={
+        English={
+          ['1']='Valaineral',
+          ['2']='Mihli Aliapoh',
+          ['3']='Tenzen',
+          ['4']='Adelheid',
+          ['5']='Joachim'},
+        Japanese={
+          ['1']='ヴァレンラール',
+          ['2']='ミリ・アリアポー',
+          ['3']='テンゼン',
+          ['4']='アーデルハイト',
+          ['5']='ヨアヒム'}}[windower.ffxi.get_info().language]},
+    wait={
+      ['aftercast']=3,
+      ['retr']=1.25,
+      ['retrall']=3},}
+  settings = config.load(defaults)
+  lang = string.lower(settings.language)
+end)
+
+windower.register_event('addon command',function(...)
+  cmd = {...}
+  if S{'save','s'}[cmd[1]] then
+    save_set(cmd[2])
+  elseif S{'check','c'}[cmd[1]] then
+    check_learned()
+  else
+    call_set(cmd[1] or 'default')
+  end
+end)
+
+function save_set(set)
+  settings.sets[set] = {}
+  local trust_ind = 0
+  for i=1,5 do
+    local trust = windower.ffxi.get_party()['p'..i]
+    if trust and trust.mob.spawn_type == 14 then
+      trust_ind = trust_ind + 1
+      settings.sets[set][tostring(trust_ind)]=trusts:with('models',trust.mob.models[1])[lang]
+    end
+  end
+  settings:save('all')
+  log('set '..set..' saved.')
+end
+
+function check_lang(entity)
+  return {japanese=entity.japanese,english=entity.english}[lang];
+end
+
+function check_limit()
+  for i,v in pairs(windower.ffxi.get_key_items()) do
+    --Trust permit,Rhapsody in..
+    limit = S{2497,2499,2501}[v] and 3 or v==2884 and 4 or v==2886 and 5 or limit or 0
+  end
+  return limit;
+end
+
+function call_trust()
+  if #queue > 0 then
+    windower.send_command('input /ma "'..windower.to_shift_jis(check_lang(queue[1]))..'" <me>')
+  end
+end
+
+function check_exist()
+  local party = {} --include only trusts. --['name']=models
+  local party_ind = {} -- index of trust's name in current party. {'name1','name2',...,'name5'}
+  for i=1,5 do
+    local member = windower.ffxi.get_party()['p'..i]
```

Comment:
Try to move `windower.ffxi.get_party()` to the outside of the loop. Write it to a local variable `local party = ...` and then use the `party` variable inside.